src/lib/tls/tls_callbacks.h

```diff
@@ -560,6 +560,18 @@ class BOTAN_PUBLIC_API(2, 0) Callbacks {
       virtual void tls_log_debug_bin(const char* descr, const uint8_t val[], size_t val_len) {
          BOTAN_UNUSED(descr, val, val_len);
       }
+
+      /**
+       * Optional callback: gives away data for ssl key log file
+       * @param label secret label for tls
+       * @param client_random random value from ClientHello message
+       * @param secret secret value
+       */
```

Comment:
> Consider renaming TLS::Callbacks::tls_ssl_key_log_data() to ...::tls_log_secret()

I disagree, because the name of this method reflects the the environment variable, which is assigned to RFS to automatically enable logging.
```
In many implementations, the file that the secrets are logged to is specified in an environment variable named "SSLKEYLOGFILE", hence the name of SSLKEYLOGFILE format.
```

Reply:
> The test cases in src/tests/test_tls_rfc8448.cpp are well-suited to verify this new callback; they feature some hard-coded TLS 1.3 traces as specified in [RFC 8448](https://www.rfc-editor.org/rfc/rfc8448). I can take this task, as it will likely require some additions to the test's already intricate architecture.

I will be very grateful

Reply:
> Optional: Add an example (in src/examples) that uses the callback to actually emit a file stream formatted as specified in draft-thomson-tls-keylogfile-00

I can do this.

Reply:
> I will be very grateful

I'll do that once we converged on the design of this patch. (Perhaps as a follow-up PR as I'll be AFK for a two weeks starting tomorrow.)